pkg/logs/diagnostic/message_receiver_test.go

```diff
@@ -9,13 +9,21 @@ import (
 
 	"github.com/stretchr/testify/assert"
 
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
 	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/util/testutils"
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
 	"github.com/DataDog/datadog-agent/pkg/logs/sources"
 )
 
+var hostname hostnameinterface.HostnameInterface
+
+func init() {
+	hostname = testutils.NewHostnameMock("unknown")
+}
```

Comment:
```suggestion
var hostname = testutils.NewHostnameMock("unknown")
```

Reply:
I got advice previously to remove global state from test files in this comment: https://github.com/DataDog/datadog-agent/pull/21987#discussion_r1450903154.

So I just updated my PR to use a helper function instead.

Reply:
It would be even better but in your code `hostname` is a global state.

Reply:
To clarify my previosu feedback, I more or less meant to do something like:

```go
b := NewBufferedMessageReceiver(nil, testutils.NewHostnameMock("unknown"))
```

or if you really want to not repeat that:

```go
func getNewHostName() hostname.Component {
    return testutils.NewHostnameMock("unknown")
} 

b := NewBufferedMessageReceiver(nil, getNewHostName())
```

Reply:
Yeah that makes sense to me, I used the second approach in my PR